### PR TITLE
Toolchian: Add nix flake; Fix nix build deps

### PR DESCRIPTION
**Pull request by Alizter**
Add a nix flake to `Toolchain/` that wraps the existing nix derivation
`Toolchain/serenity.nix`. This also comes with a lockfile making the nix
developer enviornment setup more reproducible.

We also update the documentation for "other builds" to mention the
flake.

xlibswrapper is removed from `Toolchain/serenity.nix` as it is a
wrapper package and has been deprecated. We don't use it in the build
anyway.

This fixes https://github.com/SerenityOS/serenity/issues/19286

During the build, python3 is required so we add `python3` as a build
dependency.

### CODE REVIEW

File: Toolchain/flake.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  description = "Serenity OS";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+  outputs = { self, flake-utils, nixpkgs }:
```
Comment (ADKaster) [inline Toolchain/flake.nix:8]: Does this flake file work just as well for Ladybird/ladybird.nix? I wonder if there's a way to use the same info for both of those environments. one for building the OS and running in QEMU, another for building the browser on your host and running it with Qt6

Comment (ADKaster) [reply Toolchain/flake.nix:8]: If it works for both, it might make sense to drop it into `Meta/` instead? to centralize.

Comment (Alizter) [reply Toolchain/flake.nix:8]: This was made to build the OS, however there is nothing stopping us from adding another dev shell here specifically for building Ladybird. Should be easy to do. If you want to see an example of a shell with multiple shells have a look (towards the end) at https://github.com/ocaml/dune/blob/main/flake.nix

Comment (ADKaster) [reply Toolchain/flake.nix:8]: Sure, room for improvement

Comment (Alizter) [reply Toolchain/flake.nix:8]: I've opened https://github.com/SerenityOS/serenity/pull/20890 for adding the ladybird shell.

**Pull request merged**